projects/core/src/state/state.module.ts

```diff
@@ -1,24 +1,27 @@
-import { NgModule, Optional } from '@angular/core';
-import { META_REDUCERS, StoreModule } from '@ngrx/store';
+import { NgModule } from '@angular/core';
 import { EffectsModule } from '@ngrx/effects';
+import { StoreModule } from '@ngrx/store';
 import { ConfigModule } from '../config/config.module';
-import { stateMetaReducers } from './reducers/index';
 import { defaultStateConfig } from './config/default-state-config';
-import { META_REDUCER, metaReducersFactory } from './meta-reducer';
+import { stateMetaReducers } from './reducers/index';
 
 @NgModule({
   imports: [
-    StoreModule.forRoot({}),
+    StoreModule.forRoot(
+      {},
+      {
+        runtimeChecks: {
+          strictStateImmutability: true,
```

Comment:
We initialize ngrx deep inside in our core, and customer is not able to provide his own config. 
1. There is an issue for moving ngrx initialization out of core and storefront lib module, to allow configurability.
2, Until then, we probably don't want to introduce custom config that user can't change, if it is not necessary.

Reply:
Good point